### DISCUSSION TITLE
Remote interface over WAN - what Port to forward?

### DISCUSSION THREAD

**Discussion by skycamel** [rern/rAudio#1731] [General]
I would like to be able to connect to my rAudio interface when outside the house. When I'm connected to my LAN I just give the IP-address without port number and I get the interface. But when outside the house I will have to forward a port to my RPI IP but which port?
Cheers

**Comment by fscara** [comment]
80/TCP

I suggest you to use a VPN to enter the house.
Exposing on the Internet a local application is not a wise choice, IMHO.

My LAN router has a VPN client to a VPN provider (actually to my OpenVPN access server which I manage with my private CA, kept offline) and it routes between its local LAN and the VPN subnet. 
On my android tablet I have a VPN client connected to the same VPN provider.
That way I can access all the local LAN services when I'm not directly connected to the router.
It is possible to install a VPN client on the same RPI with rAudio, so to avoid LAN routing.

f.

**Reply by skycamel** [reply]
Thank you very much for your insight. I'm a noob what VPN is concerned so if you could give me some more info how to do this I would appreciate that. I'm using Windscribe.
Cheers

**Reply by skycamel** [reply]
@fscara figured out myself.
In my router I created a VPN server to which I connect when I'm outside and I can reach whatever is on the LAN.
Thanks for the tip
Cheers,
Luc

**Reply by fscara** [reply]
> @fscara figured out myself. In my router I created a VPN server to which I connect when I'm outside and I can reach whatever is on the LAN. Thanks for the tip Cheers, Luc

I'm sorry having lost your request for more info, I don't access a computer every day.
I'm also happy that you figured out yourself, a solution you don't know how it works is not a really good solution.
Cheers
f.

**Reply by Thermionic-Emission** [reply]
Hello,

Have you looked into TwinGate? It is a Proxy that does not require any ports to be opened on your personal network. I have it running on my Pi4 rAudio server, and it streams my music to my iPhone when I am running around or out of town.

**Reply by skycamel** [reply]
Thanks for the tip, looks interesting. I already solved it by installing an OpenVPN tunnel in my router but I have an other installation without the possibility of OpenVPN in the router so your suggestion might bring a solution for that.
Cheers

**Reply by Thermionic-Emission** [reply]
I am also trying to setup wireguard to compare it against Twingate. What ports did you open for openVPN to work?

**Reply by skycamel** [reply]
Well, you don't really have to open any ports, that's the beauty of it.
In your router you go to VPN and enable OpenVPN Server. It usually uses port 1194 but you don't have to open this, it will be taken care of by OpenVPN. Choose UDP for steaming music or video, .. Use TCP for data transfer.
In the router OpenVPN you will have to create a certificate which you will have to import in your OpenVPN client on your mobile device.
Once your client is running, just address your devices in you home network with their normal IP address.
Cheers,

**Reply by Thermionic-Emission** [reply]
My router does not have a VPN option. I have just finished using a second RPi with a minimal install and it is running wireguard in a container. I had to open ports 80 and 1780 for snap client access and forward those to the Rpi that is the rAudio music server. IT is working very well. I believe I will use this over the Twingate method I mentioned earlier. This way I do not haveto rely on a third party.
